BreakinIn/Messages/ReptIn.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BreakinIn.Messages
+{
+    public class ReptIn : AbstractMessage
+    {
+        public override string _Name { get => "rept"; }
+
+        public string PERS { get; set; }
+        public string PROD { get; set; }
+        public string LANG { get; set; }
+		
+		public override void Process(AbstractEAServer context, EAClient client)
+		{
+			var mc = context as MatchmakerServer;
+            if (mc == null) return;
+			
+			var PlayerReportingPlayer = new ReptOut()
+            {
+                PERS = PERS,
+            };
+
+            var user = client.User;
+            if (user == null) return;
+
+            if (user.PersonaName == PERS)
+            {
+                Console.WriteLine(user.PersonaName + " attempted to report themselves, which is not allowed.");
+                client.SendMessage(new ReportingSelf());
+            }
+            else
+            {
+                Console.WriteLine(user.PersonaName + " has reported " + PERS + ". Both players are currently in " + user.CurrentRoom.Name.ToString() + ".");
```

Comment:
If `user.CurrentRoom` is null, this could throw an exception. You also haven't validated that both the target and user are in the same room, which would avoid doing this in the first place.

Reply:
Thanks for reviewing the code. All indention issues should be fixed once I upload the changes to my fork. I am however, not sure how I would go about validating both players being in the same room, as while I can easily get the reporter's user information, getting the other player's user information I'm stumped on as the only thing that would contain the name is PERS, but it's not possible to just reference user.CurrentRoom with just PERS alone...

I was thinking of adding a string, but how I'd go about implementing that I'm not too sure on either...

Reply:
What I'll do for now is just write the reporter's current room to the console, rather than report both users being in the same room, and if I can figure out a way to verify both users being in the same room, I'll add it to a separate pull request with some other changes so that it's not a 1 file pull request.